biom/tests/test_cli/test_table_converter.py

```diff
@@ -28,16 +29,19 @@ def setUp(self):
         self.cmd = _convert
         self.output_filepath = tempfile.NamedTemporaryFile().name
 
-        with tempfile.NamedTemporaryFile('w') as fh:
+        with tempfile.NamedTemporaryFile('w', delete=False) as fh:
             fh.write(biom1)
             fh.flush()
             self.biom_table1 = load_table(fh.name)
+            fh.close()
```

Comment:
Was it necessary to `close` here but not in the other `TestCase`s? And same here re `unlink`ing. If it's viable to just unlink within the context manager than that may be easier than the `tearDown`

Reply:
If you `close` the file before `unlink`ing within the context manager, then it works. Otherwise you get a `PermissionError` at the `unlink`ing step. Looking at the code now, I think the idea of setting `self.temporary_fh_name` in the context manager and putting the `unlink` in `tearDown` is the best solution.